app/controllers/products_controller.rb

```diff
@@ -0,0 +1,64 @@
+class ProductsController < ApplicationController
+
+  def index
+    @products = Product.all
+    @categories = Category.all
+    @products_array = []
+    @merchant_products = nil
+  end
+
+  def new
+  end
+
+  def category
+    @merchant_products = nil
+    @categories = Category.all
+    @category = Category.find(params[:category])
+    pro = @category.product_categories
```

Comment:
I think you could use an enumerable to select the product Ids that would help you clean up the code in this section